### PR TITLE
chore(CPlusPlus): add number system

### PR DESCRIPTION
**Pull request by nk412k**
**PR Checklist:**
- [x] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [ ] My addition is on refer on the language README.md file
- [ ] My addition does not have a spelling problem
- [ ] My submission has a proper and user-friendly description of the algorithm
- [ ] My submission has the time complexity of the algorithm
- [ ] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

**Other information:**

### CODE REVIEW

File: algorithms/CPlusPlus/Number system/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.vscode/
```
[the remark this answers is not in the document]
Comment (nk412k) [reply algorithms/CPlusPlus/Number system/.gitignore]: done

File: algorithms/CPlusPlus/Number system/decimal_to_hexa.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            char c='A'+(d-10);
+            num= num + c;
+        }
+        else
+        num=num + to_string(d);
+        n=n-(d*multipler);
+        multipler/=16;
+    }
+    return num;
+}
```
Comment (Arsenic-ATG) [inline algorithms/CPlusPlus/Number system/decimal_to_hexa.cpp:23]: typos detected by codespell 

```
./algorithms/CPlusPlus/Number system/decimal_to_hexa.cpp:5: multipler ==> multiplier, multiple
./algorithms/CPlusPlus/Number system/decimal_to_hexa.cpp:6: multipler ==> multiplier, multiple
./algorithms/CPlusPlus/Number system/decimal_to_hexa.cpp:7: multipler ==> multiplier, multiple
./algorithms/CPlusPlus/Number system/decimal_to_hexa.cpp:8: multipler ==> multiplier, multiple
./algorithms/CPlusPlus/Number system/decimal_to_hexa.cpp:11: multipler ==> multiplier, multiple
./algorithms/CPlusPlus/Number system/decimal_to_hexa.cpp:12: multipler ==> multiplier, multiple
./algorithms/CPlusPlus/Number system/decimal_to_hexa.cpp:19: multipler ==> multiplier, multiple
./algorithms/CPlusPlus/Number system/decimal_to_hexa.cpp:20: multipler ==> multiplier, multiple
```

```suggestion
string decimal_to_hexa(int n){
    int multiplier=1;
    while(multiplier<=n)
        multiplier*=16;
    multiplier/=16;
    string num="";
    int d;
    while(multiplier>=1){
        d=n/multiplier;
        if(d>9){
            char c='A'+(d-10);
            num= num + c;
        }
        else
        num=num + to_string(d);
        n=n-(d*multiplier);
        multiplier/=16;
    }
    return num;
}
```

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]
every addition of an algorithm should also accompany a change updating the corresponding index README.md file (CPlusPlus/README.md).

**Comment by Arsenic-ATG** [comment]
now the only thing left is updating the proper readme and this should be good to go

**Review by Arsenic-ATG — APPROVED** [review]

**Pull request merged**